Quantum-Control-Applications/Superconducting/Two-Flux-Tunable-Transmons/Use Case 4 - Single- and Two-Qubit State and Process Tomography/helper_functions.py

```diff
@@ -0,0 +1,655 @@
+# helper functions pertinent to single- and two-qubit tomography
+
+import numpy as np
+import functools
+
+import matplotlib.pyplot as plt
+from matplotlib.colors import Normalize
+from matplotlib import cm, colormaps
+
+
+titlefont = {'color':  'black', 'weight': 'normal', 'size': 10}
+axisfont = {'color':  'black', 'weight': 'normal', 'size': 8}
+ticksfont = {'color':  'black', 'weight': 'normal', 'size': 8}
+
+
+from qm.qua import *
+
+
+def rotated_multiplexed_state_discrimination(
+    I,
+    I_st,
+    Q,
+    Q_st,
+    states,
+    states_st,
+    resonators,
+    thresholds
+):
+    """
+    Perform multiplexed state discrimination on two qubits
+
+    :param I: List of QUA variables to which the I quadrature measurements
+    are written
+    :param I_st: Optional list of QUA streams to which I values are streamed
+    :param Q: List of QUA variables to which the Q quadrature measurements
+    are written
+    :param states: List of QUA variables to which qubit state are written
+    :param states_st: Optional list of QUA streams to which qubit states
+    are streamed
+    :param resonators: List of integers denoting readout resonators
+    appearing in the config file as "ff{res}" elements
+    :param thresholds: List of readout thresholds from a calibrated
+    IQ blob two-state discriminator. I values above each threshold
+    signify that the qubit was found to be in the excited state
+
+    :return:
+    """
+
+    if type(resonators) is not list:
+        resonators = [resonators]
+
+    if type(thresholds) is not list:
+        thresholds = [thresholds]
+
+    for ind, res in enumerate(resonators):
+        measure(
+            "readout",
+            f"rr{res}",
+            None,
+            dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I[ind]),
+            dual_demod.full("rotated_minus_sin", "out1", "rotated_cos", "out2", Q[ind]),
```

Comment:
Let's write it in the new SDK, i.e. : 
                    dual_demod.full("rotated_cos", "rotated_sin", I[ind]),
                    dual_demod.full("rotated_minus_sin", "rotated_cos", Q[ind]),

Reply:
Does the new SDK applies for OPX+ as well?

Reply:
What about the reverse? will this version work in the latest qm-qua versions? if that is the case then let's keep it as is, and we can revisit it in 6 months from now when more customers have transitioned to newer API and QOP :)

Reply:
Yes it is currently working the other way around. Ok :)